include/dlaf/eigensolver/bt_band_to_tridiag/mc.h

```diff
@@ -149,13 +247,16 @@ void BackTransformationT2B<Backend::MC, Device::CPU, T>::call(Matrix<T, Device::
   if (mat_hh.size().isEmpty() || mat_e.size().isEmpty())
     return;
 
-  const SizeType mb = mat_e.blockSize().rows();
-  const SizeType b = mb;
+  // Note: if no householder reflectors are going to be applied (in case of trivial matrix)
+  if (mat_hh.size().rows() <= (dlaf::isComplex_v<T> ? 1 : 2))
+    return;
+
+  const SizeType b = band_size;
   const SizeType nsweeps = nrSweeps<T>(mat_hh.size().cols());
 
   const auto& dist_i = mat_hh.distribution();
```

Comment:
I went for a generic I for INPUT, just because it is the matrix with reflectors in compact form. If you have any better suggestion, I'm ready to address that in this PR too 😉

Reply:
as the matrix is called `mat_hh` what about using `dist_hh`?